cores/esp8266/FS.cpp

```diff
@@ -66,7 +66,7 @@ int File::read() {
     return result;
 }
 
-size_t File::read(uint8_t* buf, size_t size) {
+int File::read(uint8_t* buf, size_t size) {
```

Comment:
Arduino API [Stream](https://github.com/arduino/ArduinoCore-API/blob/master/api/Stream.h) class doesn't have `read(buff, size)`. In Arduino API it is only in derived [Client](https://github.com/arduino/ArduinoCore-API/blob/master/api/Client.h) class and there the return type is int (The Client class is a mess)

Reply:
This is because of [Arduino's `Client` class](https://github.com/arduino/ArduinoCore-avr/blob/9f8d27f09f3bbd1da1374b5549a82bda55d45d44/cores/arduino/Client.h#L31L35) (this link is their current master).

This change is made necessary because in this PR I bring `int read(buffer,len)` from `Client::` to `Stream::`, and that, for the need of peekBuffer's performances.

So any class that is derived from `Stream::` and that implements `::read(buffer, len)` will have to follow Arduino's `Client::read(buffer,len)` prototype (let me call it a Fixing Change :).

Currently our `HardwareSerial` needs this change, also submoduled `SoftwareSerial` (a PR is mentioned in OP, and this PR changes the submodule link, temporarily I hope, to my own fork).
FSes need also this change because they are derived from `Stream::`.